include/adc.h

```diff
@@ -294,6 +319,26 @@ struct adc_driver_api {
 #endif
 };
 
+/**
+ * @brief Retrieve the internal reference voltage
+ *
+ * @param dev Pinter to the device structure for the driver instance.
+ *
+ * @return The internal reference, in millivolts.
+ * @retval -ENOTSUP if there is no internal reference
+ */
+static inline int adc_ref_internal(struct device *dev)
+{
+	int rv = -ENOTSUP;
+
+	const struct adc_config_info *cip
+		= (struct adc_config_info *)dev->config->config_info;
```

Comment:
`const` missing

Reply:
And this is why I prefer C++, which wouldn't have let that get past the compiler.

Done, as with all other requests.  Thanks for the review; let me know if you see anything remaining in the update.